hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndexQPSResourceAllocator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright (c) 2017 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.uber.hoodie.index.hbase;
+
+import java.io.Serializable;
+
+public interface HBaseIndexQPSResourceAllocator extends Serializable {
+
+  float getQPSFractionForPutsTime(final long numPuts, final int desiredPutsTime);
+
+  float getQPSFraction(final float desiredQPSFraction, final long numPuts);
+
+  void releaseQPSResources();
```

Comment:
Generally, a release should be accompanied with an acquire..Do the 2 get methods internally acquire qps resources ?

Reply:
Sorry, did not get your question. If you mean `does releaseQPSResources() also acquire qps fraction` - no, here it is the job of the implementation class to manage the resources acquired and implement a way to release the same when releaseQPSResources() is called.

Reply:
I mean't if getQPSFraction actually does acquire but I see you already changed the name below so we are good.